extensions/wikia/FounderEmails/events/FounderEmailsRegisterEvent.class.php

```diff
@@ -1,101 +1,52 @@
 <?php
 
 class FounderEmailsRegisterEvent extends FounderEmailsEvent {
-	public function __construct( Array $data = array() ) {
+
+	const EMAIL_CONTROLLER = 'Email\Controller\FounderNewMember';
+
+	public function __construct( Array $eventData = [] ) {
 		parent::__construct( 'register' );
-		$this->setData( $data );
+		$this->setData( $eventData );
 	}
 
-	public function enabled ( $wikiId, User $user ) {
-		if ( self::isAnswersWiki() ) {
-			return false;
-		}
+	public function enabled( $wikiId, User $user ) {
 
-		// disable if all Wikia email disabled
-		if ( $user->getBoolOption( 'unsubscribed' ) ) {
+		// don't send if we're on an answersWiki
+		if ( self::isAnswersWiki() ) {
 			return false;
 		}
 
-		// If digest mode is enabled, do not create user registration event notifications
-		if ( $user->getOption( "founderemails-complete-digest-$wikiId" ) ) {
-			return false;
-		}
-		if ( $user->getOption( "founderemails-joins-$wikiId" ) ) {
-			return true;
-		}
-		return false;
+		return true;
 	}
 
 	public function process( Array $events ) {
-		global $wgEnableAnswers, $wgCityId;
-		wfProfileIn( __METHOD__ );
-
-		if ( $this->isThresholdMet( count( $events ) ) ) {
-			// get just one event when we have more... for now, just randomly
-			$eventData = $events[rand( 0, count( $events ) -1 )];
 
-			$founderEmailObj = FounderEmails::getInstance();
-			$wikiService = ( new WikiService );
-			$user_ids = $wikiService->getWikiAdminIds();
-			$foundingWiki = WikiFactory::getWikiById( $wgCityId );
-
-			$emailParams = array(
-				'$EDITORNAME' => $eventData['data']['userName'],
-				'$EDITORPAGEURL' => $eventData['data']['userPageUrl'],
-				'$EDITORTALKPAGEURL' => $eventData['data']['userTalkPageUrl'],
-				'$WIKINAME' => $foundingWiki->city_title,
-				'$WIKIURL' => $foundingWiki->city_url,
-			);
-			$wikiType = !empty( $wgEnableAnswers ) ? '-answers' : '';
-
-			foreach ( $user_ids as $user_id ) {
-				$user = User::newFromId( $user_id );
-
-				// skip if not enable
-				if ( !$this->enabled( $wgCityId, $user ) ) {
-					continue;
-				}
-				self::addParamsUser( $wgCityId, $user->getName(), $emailParams );
-
-				$langCode = $user->getOption( 'language' );
-				$mailSubject = strtr( wfMsgExt( 'founderemails' . $wikiType . '-email-user-registered-subject', array( 'content' ) ), $emailParams );
-				$mailBody = strtr( wfMsgExt( 'founderemails' . $wikiType . '-email-user-registered-body', array( 'content' ) ), $emailParams );
+		if ( !$this->isThresholdMet( count( $events ) ) ) {
```

Comment:
In a different PR I eliminated the threshold setting and removed the ```isThresholdMet``` over-engineering.  I would just replace this with:
```php
if ( $count( $events ) > 0 ) {
...
}
```